pkg/pub_integration/test/exported_bucket_test.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+import 'package:path/path.dart' as p;
+import 'package:pub_integration/src/fake_test_context_provider.dart';
+import 'package:shelf/shelf.dart' as shelf;
+import 'package:shelf/shelf_io.dart' as shelf_io;
+import 'package:test/test.dart';
+
+void main() {
+  group(
+    'exported bucket works with pub get',
+    () {
+      late final TestContextProvider fakeTestScenario;
+      final httpClient = http.Client();
+      late final Directory temp;
+      late final HttpServer proxy;
+
+      setUpAll(() async {
+        fakeTestScenario = await TestContextProvider.start();
+        temp = await Directory.systemTemp.createTemp('exported-bucket');
+
+        final pubUri = Uri.parse(fakeTestScenario.pubHostedUrl);
+        final storageUri = pubUri.replace(
+            port: pubUri.port + 1, path: '/fake-exported-apis/latest');
+
+        // read-only proxy server with minimal content rewrite
+        proxy = await shelf_io.serve((rq) async {
+          try {
+            // the proxy url with the appropriate path prefix
+            final proxyUri = storageUri.replace(pathSegments: [
+              ...storageUri.pathSegments,
+              ...rq.requestedUri.pathSegments
+            ]);
+
+            // archive files are served from the storage bucket directly
+            if (rq.requestedUri.path.endsWith('tar.gz')) {
+              return shelf.Response.seeOther(proxyUri);
```

Comment:
this is not what GCLB does, that's what this proxy is emulating, right?

GCLB just serves directly, but yes, this is fine.

Reply:
Good point, removing this.